include/libcyphal/presentation/publisher.hpp

```diff
@@ -78,10 +78,12 @@ class PublisherBase
 
     PublisherBase& operator=(PublisherBase&& other) noexcept
     {
-        CETL_DEBUG_ASSERT(impl_ != nullptr, "Not supposed to move to already moved `this`.");
```

Comment:
Such assert is wrong - "swap" usually implemented by using move constructor and then 2 move assignments, like:
```
      _Tp __tmp = _GLIBCXX_MOVE(__a);
      __a = _GLIBCXX_MOVE(__b);
      __b = _GLIBCXX_MOVE(__tmp);
```
I found and fixed 3 such places in total.